app/main.py

```diff
@@ -1,8 +1,33 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = []
+    people_mapping = {}
+
+    for person_data in people:
+        name = person_data["name"]
+        age = person_data.get("age", None)
+        person = Person(name, age)
+        people_mapping[name] = person
+        person_list.append(person)
```

Comment:
You could directly append the Person instance, no need to create the `person` variable.